src/ensembl/utils/database/unittestdb.py

```diff
@@ -66,49 +73,75 @@ class UnitTestDB:
 
     """
 
-    def __init__(self, server_url: StrURL, dump_dir: StrPath, name: Optional[str] = None) -> None:
+    def __init__(
+        self,
+        server_url: StrURL,
+        dump_dir: StrPath | None = None,
+        name: str | None = None,
+        metadata: MetaData | None = None,
+        tmp_path: Path | None = None,
+    ) -> None:
         db_url = make_url(server_url)
-        dump_dir_path = Path(dump_dir)
-        db_name = os.environ["USER"] + "_" + (name if name else dump_dir_path.name)
+        if not name:
+            name = Path(dump_dir).name if dump_dir else "testdb"
+        db_name = os.environ["USER"] + "_" + name
+
         # Add the database name to the URL
         if db_url.get_dialect().name == "sqlite":
-            db_url = db_url.set(database=f"{db_name}.db")
+            db_path = tmp_path / db_name if tmp_path else db_name
+            db_url = db_url.set(database=f"{db_path}.db")
         else:
             db_url = db_url.set(database=db_name)
+
```

Comment:
```suggestion
```
Whilst I'm happy to leave the new lines you add for code clarity, I prefer not to add them where no other code is edited to ease change tracking.